app/models/notification.rb

```diff
@@ -434,6 +434,10 @@ def send_email
 # Indexes
 #
 #  idx_notifications_speedup_unread_count                       (user_id,notification_type) WHERE (NOT read)
+#  index_notifications_on_data_display_username                 ((((data)::jsonb ->> 'display_username'::text))) WHERE (((data)::jsonb ->> 'display_username'::text) IS NOT NULL)
+#  index_notifications_on_data_original_username                ((((data)::jsonb ->> 'original_username'::text))) WHERE (((data)::jsonb ->> 'original_username'::text) IS NOT NULL)
+#  index_notifications_on_data_username                         ((((data)::jsonb ->> 'username'::text))) WHERE (((data)::jsonb ->> 'username'::text) IS NOT NULL)
+#  index_notifications_on_data_username2                        ((((data)::jsonb ->> 'username2'::text))) WHERE (((data)::jsonb ->> 'username2'::text) IS NOT NULL)
```

Comment:
Instead of adding four new index here, I think we can update the query in `Jobs::UpdateUsername` to the following instead:

```
        WHERE COALESCE(
          data::JSONB ->> 'original_username',
          data::JSONB ->> 'display_username',
          data::JSONB ->> 'username',
          data::JSONB ->> 'username2'
        ) = :old_username
```

And instead of 4 indexes, we can have one instead.

```
CREATE INDEX CONCURRENTLY <index_name>
ON notifications (
  COALESCE(
    data::jsonb->>'original_username',
    data::jsonb->>'display_username',
    data::jsonb->>'username',
    data::jsonb->>'username2'
  )
);
```

Reply:
That's a great idea! Thanks @tgxworld.
Could you implement that change @branquinhoaa?

Reply:
I can't test it right now, but I assume the query will only use the index for `original_username` and ignore it for the rest of the columns. Unless something changed, I don't think PG uses multicolumn indexes for `OR` conditions. Maybe an index with `GIN` could help. I can test it on Wednesday with a large DB.